### PR TITLE
Add a section comparing the `std` and `no_std` approaches

### PR DESCRIPTION
**Pull request by jessebraham**
The first couple commits are essentially just formatting, spelling, and clarification related updates. Just cleaning things up a little.

The most recent commit adds a section comparing the two approaches, `std` and `no_std`. While I think I've included some decent information here, I think I've sort of missed the mark. It still seems that I'm not making a compelling case for either approach. I would love to hear some suggestions on what is missing, what should be changed, etc.

A rendered view of the section in question can be seen [here](https://github.com/esp-rs/book/blob/docs/std-vs-no-std/src/overview.md#comparing-std-and-no_std).

### CODE REVIEW

**Comment by bjoernQ** [comment]
Already looks quite good to me. Maybe a link to https://docs.rust-embedded.org/book/intro/no-std.html might be a good thing.

> It still seems that I'm not making a compelling case for either approach.

Don't know how to do this currently - it's also a quite unique situation to be able to choose on an MCU. To be honest the current restrictions on using _no_std_ barely gives a real choice (but that will change in future). Maybe this is already a good start and can be refined once there is a real choice (i.e. supporting more chips and having support for Wifi and BLE with _no_std_)

**Comment by jessebraham** [comment]
Thank you for taking a look! That link was included in the previous _Bare Metal (`no_std`)_ section already, so I think that's probably fine.

I think you're right, there's not really much else to add right now due to the limitations. Once things change this can be updated, but having this would definitely be better than having nothing for now.

File: src/overview.md [hunk 1]
```diff
@@ -1,79 +1,179 @@
-# Ecosystem overview
+# Ecosystem Overview
 
-As it stands there are two approaches for using Rust on Espressif chips, `std` and `no_std`. Both approaches have thier pros and cons, you should choose the approach based on your project's needs.
+As it stands there are two approaches for using Rust on Espressif chips, `std` and `no_std`. Both approaches have their advantages and disadvantages, so you should choose one based on your project's needs. Below is an overview of the two approaches followed by a brief comparison between them.
 
-## Using the Rust standard library
+The [esp-rs organization] on GitHub is home to a number of repositories related to running Rust on ESP. Most of the required crates have their source code hosted here.
```
Comment (MabezDev) [inline src/overview.md:5]: Should this be ESP's or perhaps Espressif chips?

File: src/introduction.md [hunk 2]
```diff
@@ -10,6 +10,15 @@ For tools and libraries relating to Rust on ESP, please see the [esp-rs organiza
 [espressif]: https://espressif.com/
 [esp-rs organization]: https://github.com/esp-rs/
 
+## Status of This Book
+
+This book is currently a work in progress. A number of sections may be missing information or be missing altogether. If there is a specific topic you would like to see documented please [open an issue].
+
+If you feel you can contribute something to this book, we encourage you to [create a pull request]!
+
+[open an issue]: https://github.com/esp-rs/book/issues
```
Comment (MabezDev) [inline src/introduction.md:19]: Nit: maybe link to https://github.com/esp-rs/book/issues/new instead, to save a click?

**Review by MabezDev — APPROVED** [review]
Couple of nitpicks, but nothing that would stop this being merged. Thanks Jesse!

**Comment by jessebraham** [comment]
Thanks for the feedback to both of you!

**Pull request merged**